### ISSUE TITLE
Fully automated bearer-token-for-user (no user interaction)

### ISSUE THREAD

**Issue by andus4n** [allegro/allegro-api#9162]
Hi.
I'm just starting with this api and i have a simple question: is it possible to obtain the "bearer-token-for-user" string without the slightest user interaction? (for a console app)
After my app is registered and accepted by allegro, i just want to read from a local config the client id and secret, send them to the oauth service and obtain the tokens. That's it - **no redirect url**, **no user input**, **no nothing**.
How can i achieve this? Any suggestions?

My guess is that i should use the device-flow, but even there somekind of user interaction is required.

**Comment by Toumash** [comment]
For `bearer-token-for-user` you'll always have a user interaction for setting up the integration first time.
Later when you have it set up you can use `refresh_token` to obtain new `acces_token` infinitely (read about refresh tokens here: https://developer.allegro.pl/tutorials/uwierzytelnianie-i-autoryzacja-zlq9e75GdIR#przedluzenie-waznosci-tokena)

**Comment by andus4n** [comment]
Understood, there's no way to do that.
Alright, so if i want **minimal human interaction** i should go with the `device flow` instead of the `authorization code flow`, right?

**Comment by Toumash** [comment]
Depends ;) On your definition of "minimal human interaction". If the goal is to click minimal numbers of buttons then the standard  authorization code flow is ok. If the goal is to click minimal number of buttons **on the machine the app is running** then you can use device flow.

In general you pick the flow using the [factors from the docs](https://developer-allegro-pl.translate.goog/tutorials/uwierzytelnianie-i-autoryzacja-zlq9e75GdIR?_x_tr_sl=pl&_x_tr_tl=en&_x_tr_hl=pl&_x_tr_pto=wapp#przedluzenie-waznosci-tokena) that says:
* [Authorization Code flow](https://developer-allegro-pl.translate.goog/tutorials/uwierzytelnianie-i-autoryzacja-zlq9e75GdIR?_x_tr_sl=pl&_x_tr_tl=en&_x_tr_hl=pl&_x_tr_pto=wapp#authorization-code-flow) - the most popular authorization method for the OAuth standard, under which the application can perform operations on behalf of the user,
* [Device flow](https://developer-allegro-pl.translate.goog/tutorials/uwierzytelnianie-i-autoryzacja-zlq9e75GdIR?_x_tr_sl=pl&_x_tr_tl=en&_x_tr_hl=pl&_x_tr_pto=wapp#device-flow) - enables authorization on devices or applications that **do not have a graphical interface** or a way to enter text,

In short:
* for something like Smart TV you use device flow cause you have a stupid remote without a real keyboard + browser (see: youtube for smarttv)
* for console apps you use authorization code flow plus device flow as the fallback if user does not have access to the browser (runs on a server)
* for normal GUI apps you just use authorization code
* for web apps you use authorization code

**Comment by Lukasz-Zurek** [comment]
@andus4n I think that **Toumash** has summed it up perfectly: 

> For bearer-token-for-user you'll always have a user interaction for setting up the integration first time.

so I can only confirm it. Then [refreshing tokens](https://developer.allegro.pl/tutorials/authentication-and-authorization-m09BlVyo7iY#extending-token-validity) by refresh_token for the user goes in a transparent way (without the need for the user to perform any actions).
